### PR TITLE
Uses petitio (zero dep http lib) instead of built-in http(s)

### PR DESCRIPTION
**Pull request by diced**

### CODE REVIEW

File: src/structures/Manager.ts [hunk 1]
```diff
@@ -225,27 +224,35 @@ export class Manager extends EventEmitter {
...
+    //   }, (res) => {
+    //     let data = Buffer.alloc(0);
+    //     res.on("data", (chunk) => data = Buffer.concat([ data, chunk ]));
+    //     res.on("error", (e) => reject(e));
+    //     res.on("end", () => resolve(JSON.parse(data.toString())));
+    //   });
+
+    //   res.on("error", e => reject(e));
+    //   res.end();
+    // });
```
Comment (MeLike2D) [inline src/structures/Manager.ts]: remove comments pls

**Review by MeLike2D — APPROVED** [review]
yes

**Comment by MeLike2D** [comment]
ok

**Comment by muhimur9049** [comment]
@diced Just curious but is this petitio package faster than any of the other HTTP modules like Axios, node-fetch, etc.?

The Lavalink library I use uses node-fetch so that's why I'm wondering if it's worth switching or not.

**Comment by melike2d** [comment]
> @diced Just curious but is this petitio package faster than any of the other HTTP modules like Axios, node-fetch, etc.?
> 
> The Lavalink library I use uses node-fetch so that's why I'm wondering if it's worth switching or not.

The github repository for petitio has benchmarks in it somewhere

**Comment by melike2d** [comment]
@muhimur9049 https://github.com/helperdiscord/http-benchmarks/blob/main/RESULTS.md

**Comment by muhimur9049** [comment]
Ah, that's cool

What's the `(undici)` version?
image

**Comment by melike2d** [comment]
> Ah, that's cool
> 
> What's the `(undici)` version?
> image

[undici](https://github.com/nodejs/undici) is a custom nodejs http library, it's made by some nodejs contributors

**Pull request merged**